app/services/service_result.rb

```diff
@@ -198,6 +198,12 @@ def message
     end
   end
 
+  def merge_dependent_errors!
```

Comment:
I've created this method in the ServiceResult class. However, it would also be no problem to keep this as a private method on Storages::Admin::StoragesController...

Reply:
I understand why this is needed. I would like to just point out the the `ServiceResult` class is doing now even more things. We should think about separating the error collection part in general. Collecting multiple errors is important so I would say keep it here for now.

Nevertheless I would suggest not having a mutating method for this but just a computed property or method that computes the flattened errors on demand. You could call it `all_errors` or `collected_errors` or something like that. Then you would just have a new property on the `ServiceResult` without mutating it again.